### PR TITLE
Add Quest: building:material

### PR DESCRIPTION
**Pull request by ravenfeld**
I'm going to split the request in 2.
https://github.com/Helium314/SCEE/issues/443

Creating a quest to add building material

Screenshot_20240703_105109

### CODE REVIEW

File: app/src/main/res/values/strings_ee.xml [hunk 1]
```diff
@@ -671,4 +671,35 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
...
+    <string name="quest_building_material_value_concrete">concrete</string>
+    <string name="quest_building_material_value_metal">metal</string>
+    <string name="quest_building_material_value_steel">steel</string>
+    <string name="quest_building_material_value_stone">stone</string>
+    <string name="quest_building_material_value_glass">glass</string>
+    <string name="quest_building_material_value_mirror">mirror</string>
+    <string name="quest_building_material_value_mud">mud</string>
+    <string name="quest_building_material_value_tin">tin</string>
+    <string name="quest_building_material_value_plastic">plastic</string>
+    <string name="quest_building_material_value_timber_framing">timber_framing</string>
```
Comment (mcliquid) [inline app/src/main/res/values/strings_ee.xml]: I would convert the strings from OSM tags to English spelling. So instead of "timber_framing", for example, make it "Timber Framing".

Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: Yes error with the cherry pick of my main branch

**Comment by HolgerJeromin** [comment]
Metal, steel and tin is a bit confusing. But yeah, all are [in use](https://taginfo.openstreetmap.org/keys/building%3Amaterial#values)...

**Comment by ravenfeld** [comment]
Yes, I said to myself that I couldn't select a value. Maybe change the message to indicate that only experts can use it?

Or other photos but I myself am not an expert on the subject.

**Comment by HolgerJeromin** [comment]
Perhaps drop `steel` and `tin` and let `metal` as the only value.

**Comment by ravenfeld** [comment]
@Claurt07 @Un1matr1x

What do you say to these tags?

I like having the choice, especially when I know the subject well. In this case, it would mean that contributors are being forced to use metal, which may or may not be the case.
But I'll let those who contribute to these values tell us what they need.

**Comment by mcliquid** [comment]
`metal` is a generic term for a variety of elements and alloys with metallic properties. `steel` is a solid alloy of iron and carbon. `steel` is defined in the norm "EN 10020". `tin` is a soft, silvery-white metal. `tin` forms verdigris (green mould), which is often found on church roofs ([Example](https://commons.wikimedia.org/wiki/File:Dome,_globe_and_verdigris_-_Trafalgar_Square_(geograph_2640800).jpg)).

As a comparison to `surface`: `metal` behaves more or less like `paved` while `steel`, `aluminium` or `tin` are more specific.

**Comment by Claurt07** [comment]
@mcliquid thanks for shedding some light on that -- I have to acknowledge I was not clear on the distinction of some of these. 
Keeping at least `metal` or `steel`  and `tin` (or the three) are good for my use case, as we provide a way for people to specify a form of metal that is generic and one that might be more "vulnerable" than the rest. 

@ravenfeld the rest looks good to me! the images are also very illustrative. 
I only have another question, should `Mud` and `Adobe`  be combined and leave only `Mud`? It is called `Adobe` where I am from in LATAM, but I wonder if it is not too specific.

**Comment by HolgerJeromin** [comment]
@mcliquid
You are right, but the target audience is probably not able to differ steel from other metals...

**Comment by mcliquid** [comment]
@HolgerJeromin I definitely agree with the target group of StreetComplete. But here we are talking about SCEE - the StreetComplete Expert Edition, right?
My understanding of this version is that we can offer quests here that can go beyond the normal expected level of knowledge, as we can expect the specific target group to intentionally disable such quests (each of which comes with an individual warning) if they are unable to answer them.

**Comment by Helium314** [comment]
I did not have a look at basically anything yet, but please massive reduce image size. This PR alone is adding ~4 megabytes to app size.

Also not sure about the general style when I look at the images. Some of the tags appear to be categories, e.g. metal -> tin+steel, or stone -> sandstone.

Btw what is `mirror` supposed to mean? "This building is made of mirrors"?

**Comment by ravenfeld** [comment]
@mcliquid @Claurt07 @Un1matr1x
Can you help me find more meaningful images? I've looked on the wiki and on wikimedia common. As this is not a quest that I would undertake and as it is taking me a long time to find images, I would like to ask for your help. 
Thank you for your help.

**Comment by mnalis** [comment]
**TL;DR:** perhaps go just with `metal` (or `metal` and `copper`), ignoring other metal-detailed answers.

---

> `tin` forms verdigris (green mould), which is often found on church roofs ([Example](https://commons.wikimedia.org/wiki/File:Dome,_globe_and_verdigris_-_Trafalgar_Square_(geograph_2640800).jpg)).

Actually, it is _**copper**_ salts that produce that green patina, not ***tin***. Tin is major component of [solder](https://en.wikipedia.org/wiki/Solder), and tin foil is what we used to use  to make [tin foil hats](https://en.wikipedia.org/wiki/Tin_foil_hat) to protect us from government (or alien) mind-reading devices (or to wrap food)  :smiley:  - although it has mostly been superseded by aluminium foil nowdays, but name stuck. 

As for building material, while I generally prefer detailed answers to be available in SCEE, I'm not so sure here. 

Basically, **only copper is distinctive, other metals not** so much.  So I'd be fine with just `metal` (or `metal` and `copper`, but that is bound to have people asking questions _"why is there copper, and not tin"_ etc).

---

Most people (who are not material scientists _or_ are unwilling to scratch the surfaces to determine what it is) will be hard pressed to tell the difference between [corrugated galvanized iron](https://en.wikipedia.org/wiki/Corrugated_galvanised_iron) (sometimes called "wriggly tin" to add to the confusion, even if Zinc is used for galvanization), steel that has been coated with Zinc-Aluminium,  iron/steel coated with thin layer of tin (also known as "[Tinplate](https://en.wikipedia.org/wiki/Tinplate)"), various aluminium alloys (aluminium+manganese, aluminium+magnesium etc.) and steel coated with alloy of lead and tin (also called Tinplate colloquially, but more precisely [Terne](https://en.wikipedia.org/wiki/Terne), even if in original coating there was order of magnitude more lead then tin; but the coating is not load bearing (majority) material anyway). 

The use of actual `tin` as building/roof material is not what one would find in real life, it is too structurally weak. Also, "Tinplates" often do not contain `tin` at all (and even when they do, it is in micro amounts only for coating, main load bearing material usually being steel; but there are also various aluminum alloys which people might incorrectly call by such "tin" names because of history - just like Tinfoil hat above)

**Comment by mcliquid** [comment]
> So I'd be fine with just `metal`

I agree, let's start simple and pick only the significant values. You can always extend a quest later!

**Comment by ravenfeld** [comment]
What do you want me to display then?

**Comment by mnalis** [comment]
> What do you want me to display then?

Drop `steel`, `tin`, `copper`, `metal_plates` (but retain generic `metal`)

**Comment by mnalis** [comment]
As far as I can see, those are several issues mentioned so far that prevent this PR:

- update to avoid `strings_ee.xml` conflicts
- update `EE_QUEST_OFFSET + 48` to new value after updating to latest master
- [please massive reduce image size. This PR alone is adding ~4 megabytes to app size](https://github.com/Helium314/SCEE/pull/569#issuecomment-2211623724)

File: app/src/main/java/de/westnordost/streetcomplete/quests/building_material/AddBuildingMaterial.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class AddBuildingMaterial : OsmFilterQuestType<BuildingMaterial>() {
+
+    override val elementFilter = """
+        ways, relations with
+          ((building and building !~ no|construction|roof|carport)
+          or (building:part and building:part !~ no|construction|roof|carport))
+          and !building:material
+          and (!indoor or indoor = no)
+          and wall !~ no
```
Comment (mnalis) [inline app/src/main/java/de/westnordost/streetcomplete/quests/building_material/AddBuildingMaterial.kt]:
```suggestion
          and indoor != no
          and wall != no
```

File: app/src/main/res/values/strings_ee.xml [hunk 3]
```diff
@@ -671,4 +671,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
     <!-- actually there are still SC translations available, as only the main string has been removed from SC. -->
     <string name="close">Close</string>
 
+    <!-- building material -->
+    <string name="quest_buildingMaterial_title">"What overall material does this surface material of building walls have?"</string>
+    <string name="quest_buildingPartMaterial_title">"What overall material does this building part have?"</string>
+    <string name="default_disabled_msg_buildingMaterial">This quest type is disabled by default because this quest type is also quite time-consuming and requires expertise</string>
```
Comment (mnalis) [inline app/src/main/res/values/strings_ee.xml]: Unless the text is going to specific to only that quest, it should either:
- named generically so can be reused (e.g. `default_disabled_msg_expertise_needed`)
- if extra information is not needed, perhaps use already existing `default_disabled_msg_difficult_and_time_consuming` instead?

File: app/src/main/res/values/strings_ee.xml [hunk 4]
```diff
@@ -671,4 +671,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
     <!-- actually there are still SC translations available, as only the main string has been removed from SC. -->
     <string name="close">Close</string>
 
+    <!-- building material -->
+    <string name="quest_buildingMaterial_title">"What overall material does this surface material of building walls have?"</string>
+    <string name="quest_buildingPartMaterial_title">"What overall material does this building part have?"</string>
```
Comment (mnalis) [inline app/src/main/res/values/strings_ee.xml]:
```suggestion
    <string name="quest_buildingMaterial_title">"What is the overall surface material of building walls here?"</string>
    <string name="quest_buildingPartMaterial_title">"What overall material is this building part made of?"</string>
```

perhaps something along those lines would be clearer? Also, is it intentional that `building` and `building:part` questions have different meaning - i.e. (bearing) material vs. surface material?

Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: I'm very bad at English.

File: app/src/main/res/values/strings_ee.xml (hunk 3 above)
Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: I have the impression that all the quests have a text, so I did the same. Maybe it's the text that needs to be changed?

Comment (Helium314) [reply app/src/main/res/values/strings_ee.xml]: > I have the impression that all the quests have a text

All SCEE quests have a default disabled message, but many use the generic SCEE message.

**Comment by ravenfeld** [comment]
Hello, I'm coming back to my PR. I've used my version a lot, which I've modified, and it's not very good for the community. I'm going to make one PR by one PR to avoid having too many open with you.

I've used ImageOptim (https://imageoptim.com/versions.html
) to compress images, it's simple and local.

File: app/src/main/res/values/strings_ee.xml (hunk 3 above)
Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: I used R.string.default_disabled_msg_difficult_and_time_consuming

**Comment by Helium314** [comment]
> I've used ImageOptim (https://imageoptim.com/versions.html ) to compress images, it's simple and local.

Thanks, that's a considerable reduction in size!

File: app/src/main/res/values/strings_ee.xml [hunk 5]
```diff
@@ -735,4 +735,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
     <string name="outside_downloaded_area_warning">This position is outside the downloaded area. Continue?</string>
     <!-- actually there are still SC translations available, as only the main string has been removed from SC. -->
     <string name="close">Close</string>
+
+    <!-- building material -->
+    <string name="quest_buildingMaterial_title">"What is the overall surface material of building walls here?"</string>
+    <string name="quest_buildingPartMaterial_title">"What overall material is this building part made of?"</string>
```
Comment (Helium314) [inline app/src/main/res/values/strings_ee.xml]: Shouldn't the building part title also refer to surface material?

File: app/src/main/res/values/strings_ee.xml [hunk 6]
```diff
@@ -735,4 +735,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
...
+    <string name="quest_building_material_value_cement_block">Cement block</string>
+    <string name="quest_building_material_value_brick">Brick</string>
+    <string name="quest_building_material_value_plaster">Plaster</string>
+    <string name="quest_building_material_value_wood">Wood</string>
+    <string name="quest_building_material_value_concrete">Concrete</string>
+    <string name="quest_building_material_value_metal">Metal</string>
+    <string name="quest_building_material_value_stone">Stone</string>
+    <string name="quest_building_material_value_glass">Glass</string>
+    <string name="quest_building_material_value_mirror">Mirror</string>
+    <string name="quest_building_material_value_mud">Mud</string>
```
Comment (Helium314) [inline app/src/main/res/values/strings_ee.xml]: Probably should be named "dried mud" to be more clear.

File: app/src/main/res/values/strings_ee.xml [hunk 7]
```diff
@@ -735,4 +735,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
...
+    <string name="quest_buildingPartMaterial_title">"What overall material is this building part made of?"</string>
+    <string name="quest_building_material_value_cement_block">Cement block</string>
+    <string name="quest_building_material_value_brick">Brick</string>
+    <string name="quest_building_material_value_plaster">Plaster</string>
+    <string name="quest_building_material_value_wood">Wood</string>
+    <string name="quest_building_material_value_concrete">Concrete</string>
+    <string name="quest_building_material_value_metal">Metal</string>
+    <string name="quest_building_material_value_stone">Stone</string>
+    <string name="quest_building_material_value_glass">Glass</string>
+    <string name="quest_building_material_value_mirror">Mirror</string>
```
Comment (Helium314) [inline app/src/main/res/values/strings_ee.xml]: Probably should be named "mirrored glass" to be more clear.

File: app/src/main/res/values/strings_ee.xml [hunk 8]
```diff
@@ -735,4 +735,31 @@ Out of the box SCEE is configured to behave very similar to StreetComplete.</str
...
+    <string name="quest_building_material_value_mud">Mud</string>
+    <string name="quest_building_material_value_plastic">Plastic</string>
+    <string name="quest_building_material_value_timber_framing">Timber Framing</string>
+    <string name="quest_building_material_value_sandstone">Sandstone</string>
+    <string name="quest_building_material_value_clay">Clay</string>
+    <string name="quest_building_material_value_reed">Reed</string>
+    <string name="quest_building_material_value_loam">Loam</string>
+    <string name="quest_building_material_value_marble">Marble</string>
+    <string name="quest_building_material_value_slate">Slate</string>
+    <string name="quest_building_material_value_vinyl">Vinyl</string>
```
Comment (Helium314) [inline app/src/main/res/values/strings_ee.xml]: The wiki rather explicitly calls this "plastic vinyl strips". Not sure though whether this needs to be mentioned in the text.

**Comment by Helium314** [comment]
This looks mostly fine now, only maybe a few details left, and @Claurt07's question is still open:
> I only have another question, should `Mud` and `Adobe` be combined and leave only `Mud`? It is called `Adobe` where I am from in LATAM, but I wonder if it is not too specific.

**Comment by mnalis** [comment]
@ravenfeld this SCEE PR (and few others by you) are almost finished and need just a little attention from you to make them mergable. Would you perhaps have time to do that in nearby future? :heart:

**Comment by ravenfeld** [comment]
I'm going to try it this weekend but with the application I'm developing for Panorama + work, I'd moved on to something other than SCEE. Sorry for the inconvenience.

**Comment by mnalis** [comment]
>I'm going to try it this weekend but with the application I'm developing for Panorama + work, I'd moved on to something other than SCEE. Sorry for the inconvenience.

No worries @ravenfeld, life priorities change; it is understood. Thank you for your contributions so far, and any that may happen on the weekend should you manage it! :heart:  

If situation changes, we'll love to have to you contribute again; but in the meantime, given the explanation, I'll assume it is fine with you for someone else interested to pick up those half-finished PRs (after the weekend) and bring them to completion?

**Comment by ravenfeld** [comment]
> > I'm going to try it this weekend but with the application I'm developing for Panorama + work, I'd moved on to something other than SCEE. Sorry for the inconvenience.
> 
> No worries @ravenfeld, life priorities change; it is understood. Thank you for your contributions so far, and any that may happen on the weekend should you manage it! ❤️
> 
> If situation changes, we'll love to have to you contribute again; but in the meantime, given the explanation, I'll assume it is fine with you for someone else interested to pick up those half-finished PRs (after the weekend) and bring them to completion?

It's not my property. Anyone can take over the PR without any worries

File: app/src/main/res/values/strings_ee.xml (hunk 5 above)
Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: I didn't understand the remark.

Comment (Helium314) [reply app/src/main/res/values/strings_ee.xml]: `quest_buildingMaterial_title` asks for _overall **surface** material_, while `quest_buildingPartMaterial_title` asks for _overall material_.
Is this difference intentional, or an oversight?

Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: No, it's just that I'm bad at English.

Comment (ravenfeld) [reply app/src/main/res/values/strings_ee.xml]: I modified it but I'm still not good at English.

**Comment by Helium314** [comment]
I think this is fine now, thank you @ravenfeld.
I just want to go through everything once more, and test the quest on my phone (hopefully tomorrow).

**Comment by Helium314** [comment]
I think it's mostly fine, only minor comment
* In some cases the images don't help much in my opinon (e.g. marble), but the text is clear enough. No need to change anything.
* You changed the text from _overall material_ for both, but on reading the wiki again, _overall surface material_ would be a better fit. Sorry for that...
Or does anyone have a better idea than _What overall surface material is this building (part) made of_?
* For the icon I think it would be nice to have something like a brick wall instead of just a plain green surface on the building.
I guess I should be able to do it if you don't think it's worth the effort.

**Comment by Helium314** [comment]
Oh, one more thing: could you rename the strings to use `material` instead of `building_material`? Then we can re-use some of them for the roof material quest.
I just renamed the strings for bench material, so we can also re-use those. No point in having the same strings multiple times, especially considering that the existing ones are already translated to many languages.

**Comment by ravenfeld** [comment]
@Helium314 I have updated what I understood from your requests.

**Comment by Helium314** [comment]
Thanks a lot, looks good!
Does anyone else have comments? Otherwise I'll merge it soon.
